updated CSP values in env.default

danielfmiranda opened the pull request:
Closes #1435 
Related PRs/issues #

## Checklist

**Changes in Models:**

- [x] [Are my changes backward-compatible]

Pomax commented on env.default:
```diff
@@ -58,6 +58,15 @@ BASKET_SQS_QUEUE_URL=
+# CSP config
+CSP_CONNECT_SRC=" * "
+CSP_FONT_SRC=" 'self' fonts.googleapis.com fonts.gstatic.com"
+CSP_FORM_ACTION=" 'self' https://webto.salesforce.com/ mozillasalescloud--mkl.cs4.my.salesforce.com "
```
Do we need both of these for the code to run correctly on localhost?

Pomax commented on env.default:
```diff
@@ -58,6 +58,15 @@ BASKET_SQS_QUEUE_URL=
+CSP_CONNECT_SRC=" * "
+CSP_FONT_SRC=" 'self' fonts.googleapis.com fonts.gstatic.com"
+CSP_FORM_ACTION=" 'self' https://webto.salesforce.com/ mozillasalescloud--mkl.cs4.my.salesforce.com "
+CSP_FRAME_SRC=" 'self' assets.braintreegateway.com c.paypal.com *.paypal.com https://www.google.com/recaptcha/api2/ https://tst.kaptcha.com/ "
```
Do we need all of these for the code to run correctly on localhost? Also some of these don't have a protocol, let's make sure everything that is necessary to run locally also starts with https://

Pomax commented on env.default:
```diff
@@ -58,6 +58,15 @@ BASKET_SQS_QUEUE_URL=
+CSP_FONT_SRC=" 'self' fonts.googleapis.com fonts.gstatic.com"
+CSP_FORM_ACTION=" 'self' https://webto.salesforce.com/ mozillasalescloud--mkl.cs4.my.salesforce.com "
+CSP_FRAME_SRC=" 'self' assets.braintreegateway.com c.paypal.com *.paypal.com https://www.google.com/recaptcha/api2/ https://tst.kaptcha.com/ "
+CSP_IMG_SRC=" * data: "
```
Do we use data: urls for images on donate? If so, then this makes sense, but do we?

Pomax commented on env.default:
```diff
@@ -58,6 +58,15 @@ BASKET_SQS_QUEUE_URL=
+CSP_FORM_ACTION=" 'self' https://webto.salesforce.com/ mozillasalescloud--mkl.cs4.my.salesforce.com "
+CSP_FRAME_SRC=" 'self' assets.braintreegateway.com c.paypal.com *.paypal.com https://www.google.com/recaptcha/api2/ https://tst.kaptcha.com/ "
+CSP_IMG_SRC=" * data: "
+CSP_SCRIPT_SRC=" 'self' 'unsafe-inline' https://www.google-analytics.com/analytics.js  https://www.google-analytics.com/plugins/ua/ecommerce.js js.braintreegateway.com assets.braintreegateway.com www.paypalobjects.com c.paypal.com www.paypal.com https://www.google.com/recaptcha/api.js https://www.gstatic.com/recaptcha/releases/" 
```
Most of these look sensible, although we want some https:// prefixes for those bare domains.

Pomax commented on env.default:
```diff
@@ -58,6 +58,15 @@ BASKET_SQS_QUEUE_URL=
+CSP_FRAME_SRC=" 'self' assets.braintreegateway.com c.paypal.com *.paypal.com https://www.google.com/recaptcha/api2/ https://tst.kaptcha.com/ "
+CSP_IMG_SRC=" * data: "
+CSP_SCRIPT_SRC=" 'self' 'unsafe-inline' https://www.google-analytics.com/analytics.js  https://www.google-analytics.com/plugins/ua/ecommerce.js js.braintreegateway.com assets.braintreegateway.com www.paypalobjects.com c.paypal.com www.paypal.com https://www.google.com/recaptcha/api.js https://www.gstatic.com/recaptcha/releases/" 
+CSP_STYLE_SRC=" 'self' 'unsafe-inline' fonts.googleapis.com fonts.gstatic.com "
```
Let's add some https:// here, too

danielfmiranda commented:
Hi Pomax, 
As discussed in our call earlier, I have updated the env.default file to match the variable names in default.py, and also added  a comment to let readers know that they can find the default values in the .py file. Please let me know if there are any more requested changes or questions. 
Thanks!

Pomax commented on env.default:
```diff
@@ -58,6 +58,18 @@ BASKET_SQS_QUEUE_URL=
 
+# CSP config
+# Default values for CSP items can be found in "donate/settings/defaults.py"
+CSP_DEFAULT_SRC=
+CSP_SCRIPT_SRC=
+CSP_IMG_SRC=
+CSP_FONT_SRC=
+CSP_STYLE_SRC=
+CSP_FRAME_SRC=
+CSP_CONNECT_SRC=
+CSP_BASE_URI=
+CSP_WORKER_SRC=
```
we don't need these empty strings now. 
```suggestion
```

danielfmiranda replied:
Hi Pomax! Thanks for the heads up, I have removed them, and resubmitted for review 👍

Pomax commented:
Sorry for the run-around! Turns out that the fix was actually just to make sure the "where to look" instructions are in a place where folks look first.
